src/cpp/database/database.cpp

```diff
@@ -4687,8 +4683,14 @@ void Database::change_entity_status_of_kind(
             {
                 topic->active = active;
 
-                details::StatisticsBackendData::get_instance()->on_domain_entity_discovery(domain_id, entity_id,
-                        entity_kind, get_status(active));
+                // TODO (eProsima) Workaround to avoid deadlock if callback implementation requires taking the database
+                // mutex (e.g. by calling get_info). A refactor for not calling on_domain_entity_discovery from within
+                // this function would be required.
+                execute_without_lock([=]()
```

Comment:
For all of them, I think `&` is better here
```suggestion
                execute_without_lock([&]()
```

Reply:
:+1: